packages/compass-e2e-tests/tests/data-modeling-tab.test.ts

```diff
@@ -9,16 +9,64 @@ import {
 import type { Compass } from '../helpers/compass';
 import * as Selectors from '../helpers/selectors';
 import {
-  createNestedDocumentsCollection,
   createNumbersCollection,
+  createNumbersStringCollection,
 } from '../helpers/insert-data';
+import {
+  cleanUpDownloadedFile,
+  waitForFileDownload,
+} from '../helpers/downloads';
+import { readFileSync } from 'fs';
 
 type DiagramInstance = {
   getNodes: () => Array<{
     id: string;
   }>;
 };
 
+async function setupDiagram(
+  browser: CompassBrowser,
+  options: {
+    diagramName: string;
+    connectionName: string;
+    databaseName: string;
+  }
+) {
+  await browser.navigateToDataModeling();
+
+  // Click on create new data model button
+  await browser.clickVisible(Selectors.CreateNewDataModelButton);
+
+  // Fill in model details
+  await browser.setValueVisible(
+    Selectors.CreateDataModelNameInput,
+    options.diagramName
+  );
+  await browser.clickVisible(Selectors.CreateDataModelConfirmButton);
+
+  // Select existing connection
+  await browser.selectOption(
+    Selectors.CreateDataModelConnectionSelector,
+    options.connectionName
+  );
+  await browser.clickVisible(Selectors.CreateDataModelConfirmButton);
+
+  // Select a database
+  await browser.selectOption(
+    Selectors.CreateDataModelDatabaseSelector,
+    options.databaseName
+  );
+  await browser.clickVisible(Selectors.CreateDataModelConfirmButton);
+
+  // TODO: Confirm all collections are selected by default (COMPASS-9309)
+  // Note: We'll need to change the UI, right now the labels are disconnected from the checkboxes
+  await browser.clickVisible(Selectors.CreateDataModelConfirmButton);
```

Comment:
You can count the checkboxes or check the `<number> total collections selected` text instead of waiting for this refactor